### PR TITLE
Port Losses to Keras Core

### PR DESCRIPTION
**Pull request by ianstenbit**
In the process, also ported related ops which are required (e.g. bounding box conversion + IoU)

### CODE REVIEW

File: keras_cv/losses/focal_test.py [hunk 1]
```diff
@@ -68,13 +55,15 @@ def test_output_shape_from_logits(self):
         )
 
     def test_from_logits_argument(self):
-        y_true = tf.random.uniform((2, 8, 10))
-        y_logits = tf.random.uniform((2, 8, 10), minval=-1000, maxval=1000)
-        y_pred = tf.nn.sigmoid(y_logits)
+        np.random.seed(1337)
+        y_true = np.random.uniform(size=(2, 8, 10))
+        y_logits = np.random.uniform(size=(2, 8, 10), low=-1000, high=1000)
```
Comment (tirthasheshpatel) [inline keras_cv/losses/focal_test.py]: Rather than setting the global seed (now legacy in NumPy), we can now do:

```suggestion
        rng = np.random.default_rng(1337)
        y_true = rng.uniform(size=(2, 8, 10))
        y_logits = rng.uniform(low=-1000, high=1000, size=(2, 8, 10))
```

File: keras_cv/losses/iou_loss_test.py [hunk 2]
```diff
@@ -12,31 +12,24 @@
...
 class IoUTest(tf.test.TestCase):
     def test_output_shape(self):
-        y_true = tf.random.uniform(
-            shape=(2, 2, 4), minval=0, maxval=10, dtype=tf.int32
-        )
-        y_pred = tf.random.uniform(
-            shape=(2, 2, 4), minval=0, maxval=20, dtype=tf.int32
-        )
+        y_true = np.random.uniform(size=(2, 2, 4), low=0, high=10)
+        y_pred = np.random.uniform(size=(2, 2, 4), low=0, high=20)
```
Comment (tirthasheshpatel) [inline keras_cv/losses/iou_loss_test.py:24]: Not related to the port so feel free to ignore but it would be good to also seed these tests.

Comment (ianstenbit) [reply keras_cv/losses/iou_loss_test.py:24]: What do we get from seeding tests where the assertions are just about shape?

Comment (tirthasheshpatel) [reply keras_cv/losses/iou_loss_test.py:24]: Only a suggestion to make the tests reproducible.

In very rare cases, for some random inputs, the test might fail making it sporadically failing through our CI runs. Seeding tests is just a way to avoid those issues. But if we are confident enough that the functions being tested shouldn't do something like that, we can keep them as-is.

Comment (ianstenbit) [reply keras_cv/losses/iou_loss_test.py:24]: Ack -- these should be resilient to any inputs within the confines of the random distribution, so I think it's fine as-is, but do let me know if you find other places where we'd be better off seeding tests 😄

Comment (tirthasheshpatel) [reply keras_cv/losses/iou_loss_test.py:24]: Sounds good, thanks for the clarification!

**Review by tirthasheshpatel — APPROVED** [review]
Looks good now, thanks for porting these!
